Drone/drone_base.py

```diff
@@ -0,0 +1,361 @@
+# Standard Library
+import abc
+import coloredlogs
+import dronekit
+from dronekit import VehicleMode
+import logging
+import math
+import time
+import threading
+
+# Ours
+from ..Utilities import constants as c
+from ..Utilities import dronekit_wrappers as dkw
+
+class DroneBase(object):
+    """
+    Wraps a DroneKit.vehicle and sensors
+
+    Data Members
+    ----------
+    vehicle: DroneKit.Vehicle
+        Interface to see parameters and send commands
+    devices: List of Device (Device not yet implemented)
+        Interface to the drone's devices, such as camera
+    connected: Boolean
+        True if the drone is connected to Ardupilot
+    """
+    __metaclass__ = abc.ABCMeta
+
+    cnt = 0
+
+    def __init__(self):
+        self.vehicle = None
+        self.devices = []
+        self.connected = False
+        self.logger = logging.getLogger(__name__)
+
+    def connect(self, isInSimulator):
+        """
+        Attempts to connect to Ardupilot
+
+        Parameters
+        ----------
+        isInSimulator: Boolean
+            True if the program is intended to run in the Gazebo simulator
+            and False if the program is being run on the real-life drone
+
+        Precondition:
+        ----------
+        None
+
+        Postcondition:
+        ----------
+        Upon a successful connection, self.connected is set to True.
+        The drone is now suitable to be armed.
+
+        Returns:
+        ----------
+        None
+        """
```

Comment:
The connection string should not be a constant here. See http://python.dronekit.io/guide/connecting_vehicle.html. The relevant quote is:

>You can control multiple vehicles from within a single script by calling connect() for each vehicle with the appropriate connection strings.

For future-proofing this code a bit, I'd recommend having a list or map of the strings for each drone and retrieving the connection string by drone id perhaps.

Reply:
Also, the default timeout on connect is 30 seconds, consider making a constant for this timeout. See http://python.dronekit.io/automodule.html#dronekit.connect  and ```heartbeat_timeout```

Reply:
Would it be appropriate for the map to be defined in the constants file or in drone_base.py?